src/ui/hooks/useDarkMode.ts

```diff
@@ -0,0 +1,20 @@
+import { useSyncExternalStore } from "react";
+
+function subscribeToPrefersColorScheme(callback: () => void): () => void {
+    const mediaQuery = window.matchMedia("(prefers-color-scheme: dark)");
+    mediaQuery.addEventListener("change", callback);
+    return () => mediaQuery.removeEventListener("change", callback);
+}
+
+function getPrefersDarkMode(): boolean {
+    return window.matchMedia("(prefers-color-scheme: dark)").matches;
+}
+
+function getServerSnapshot(): boolean {
+    return false;
+}
+
+export function useDarkMode(override?: boolean): boolean {
+    const prefersDarkMode = useSyncExternalStore(subscribeToPrefersColorScheme, getPrefersDarkMode, getServerSnapshot);
```

Comment:
Do we expect to define `getServerSnapshot` in the future? Or should this just be empty? 

```suggestion
    const prefersDarkMode = useSyncExternalStore(subscribeToPrefersColorScheme, getPrefersDarkMode);
```